### PR TITLE
set logger level for package

### LINKED ISSUE #1503: Do not configure python logging
## 🐛 Bug

pytorch-lightning right now configures the python logging module ([here](https://github.com/PyTorchLightning/pytorch-lightning/blob/8322f1b039c890b8ccdbfe29bf42056e5273d74f/pytorch_lightning/__init__.py#L16)). This is generally not recommended when writing a library as it makes it difficult for users to modify logging format (see [python docs](https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library), Stack Overflow [post](https://stackoverflow.com/questions/27016870/how-should-logging-be-used-in-a-python-package)). I would suggest deleting the configuration line.

### PR DESCRIPTION
**Pull request by jeremyjordan**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes #1503 

If a user isn't setting up any logging config on their own, we still do it for them.
If they want to set up their own logging config, this approach is less obtrusive. 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/test_profiler.py [hunk 1]
```diff
@@ -165,6 +167,8 @@ def test_advanced_profiler_describe(tmpdir, advanced_profiler):
     """
     ensure the profiler won't fail when reporting the summary
     """
+    trainer = Trainer()  # logging is set up when a Trainer object is initialized
```
Comment (Borda) [inline tests/test_profiler.py:170]: what about model or any other tool except Trainer?

**Comment by awaelchli** [comment]
@jeremyjordan will logging before the Trainer init still work, say when we would setup callbacks, loggers etc? We would just need to make sure to import Trainer before all of that, right?

**Comment by jeremyjordan** [comment]
the other option which i present [here](https://github.com/PyTorchLightning/pytorch-lightning/issues/1503#issuecomment-614343550) was to only add a stream handler for the lightning logger. i don't have a strong opinion regarding which way we handle it and am happy to switch this out for the alternative.

**Comment by Borda** [comment]
@tgaddair this is also just caching/pip issue?
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.7/x64/bin/horovodrun", line 21, in <module>
    run_commandline()
  File "/opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/site-packages/horovod/run/run.py", line 876, in run_commandline
    _run(args)
  File "/opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/site-packages/horovod/run/run.py", line 844, in _run
    _launch_job(args, remote_host_names, settings, common_intfs, command)
  File "/opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/site-packages/horovod/run/run.py", line 867, in _launch_job
    gloo_run(settings, remote_host_names, common_intfs, env, driver_ip, command)
  File "/opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/site-packages/horovod/run/gloo_run.py", line 287, in gloo_run
    _launch_jobs(settings, env, host_alloc_plan, remote_host_names, run_command)
  File "/opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/site-packages/horovod/run/gloo_run.py", line 259, in _launch_jobs
    .format(name=name, code=exit_code))
RuntimeError: Gloo job detected that one or more processes exited with non-zero status, thus causing the job to be terminated. The first process to do so was:
Process name: 1
Exit code: 1
```

**Comment by tgaddair** [comment]
No, in this case it looks like something in this PR broke rank 1:

```
Mon May 11 19:20:09 2020[0]<stderr>:2 | c_d1_drop | Dropout     | 0      | [5, 1000] | [5, 1000]
Mon May 11 19:20:09 2020[0]<stderr>:3 | c_d2      | Linear      | 10 K   | [5, 1000] | [5, 10]  
Mon May 11 19:20:09 2020[0]<stderr>:/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/utilities/distributed.py:23: UserWarning: The dataloader, val dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` in the `DataLoader` init to improve performance.
Mon May 11 19:20:09 2020[0]<stderr>:  warnings.warn(*args, **kwargs)
Mon May 11 19:20:09 2020[0]<stderr>:/home/runner/work/pytorch-lightning/pytorch-lightning/pytorch_lightning/utilities/distributed.py:23: UserWarning: The dataloader, train dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` in the `DataLoader` init to improve performance.
Mon May 11 19:20:09 2020[0]<stderr>:  warnings.warn(*args, **kwargs)
Mon May 11 19:20:11 2020[1]<stderr>:Traceback (most recent call last):
Mon May 11 19:20:11 2020[1]<stderr>:  File "/home/runner/work/pytorch-lightning/pytorch-lightning/tests/models/data/horovod/train_default_model.py", line 61, in <module>
Mon May 11 19:20:11 2020[1]<stderr>:    run_test_from_config(json.loads(args.trainer_options))
```

**Comment by Borda** [comment]
> No, in this case it looks like something (possibly a race condition) is causing rank 1 to fail:

but `in_features` in defaults and it is filing kind of randomly... any idea what could go wrong?

**Comment by tgaddair** [comment]
> > No, in this case it looks like something (possibly a race condition) is causing rank 1 to fail:
> 
> but `in_features` in defaults and it is filing kind of randomly... any idea what could go wrong?

My suspicion is that it's hitting this code path when loading the model: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/lightning.py#L1545

That suggests that the hparams haven't been updated in the checkpoint yet, which may be a race condition (rank 1 gets to this point before rank 0 has had a chance to write out the hparams).  I'll look into this further and hopefully put together a quick PR.

**Comment by Borda** [comment]
> That suggests that the hparams haven't been updated in the checkpoint yet, which may be a race condition (rank 1 gets to this point before rank 0 has had a chance to write out the hparams). I'll look into this further and hopefully put together a quick PR.

cool, Thx, as you know sooner, better because it causes difficult time for all PRs :]

**Comment by tgaddair** [comment]
> > That suggests that the hparams haven't been updated in the checkpoint yet, which may be a race condition (rank 1 gets to this point before rank 0 has had a chance to write out the hparams). I'll look into this further and hopefully put together a quick PR.
> 
> cool, Thx, as you know sooner, better because it causes difficult time for all PRs :]

I haven't been able to repro the issue locally, but my best guess at this point is still that there's a race condition, in which case https://github.com/PyTorchLightning/pytorch-lightning/pull/1786 should be able to address it.

**Comment by Borda** [comment]
> I haven't been able to repro the issue locally, but my best guess at this point is still that there's a race condition, in which case #1786 should be able to address it.

Neither me and also CircleCI does not suffer from this issue at all so I guess that it is somehow connected with GH action...

**Pull request merged**